app/javascript/crayons/MarkdownToolbar/markdownSyntaxFormatters.js

```diff
@@ -24,6 +26,40 @@ const MARKDOWN_LINK_REGEX =
   /^\[([\w\s\d]*)\]\((url|(https?:\/\/[\w\d./?=#]+))\)$/;
 const URL_PLACEHOLDER_TEXT = 'url';
 
+const NUMBER_OF_NEW_LINES_BEFORE_BLOCK_SYNTAX = 2;
+const NUMBER_OF_NEW_LINES_BEFORE_AFTER_SYNTAX = 1;
+
+const getNewLinePrefixSuffixes = ({ selectionStart, selectionEnd, value }) => {
+  const numberOfNewLinesBeforeSelection = getNumberOfNewLinesPrecedingSelection(
+    { selectionStart, value },
+  );
+  const numberOfNewLinesFollowingSelection =
+    getNumberOfNewLinesFollowingSelection({ selectionEnd, value });
+
+  // We only add new lines if we're not at the beginning of the text area
+  const numberOfNewLinesNeededAtStart =
+    selectionStart === 0
+      ? 0
+      : NUMBER_OF_NEW_LINES_BEFORE_BLOCK_SYNTAX -
+        numberOfNewLinesBeforeSelection;
+
+  let newLinesPrefix = '';
+  // only add new lines if we're not at the beginning of the text area
+  if (numberOfNewLinesNeededAtStart > 0) {
+    for (let i = 0; i < numberOfNewLinesNeededAtStart; i++) {
+      newLinesPrefix = `${newLinesPrefix}\n`;
+    }
+  }
```

Comment:
Refactor (blocking): Use the native `String.prototype.start()`

```suggestion
  const newLinesPrefix = String.prototype.padStart(numberOfNewLinesNeededAtStart + 1, '\n');
```

Reply:
Oh awesome - TIL! 🌟